library/ssl_tls13_generic.c

```diff
@@ -908,13 +908,46 @@ static int ssl_tls13_select_sig_alg_for_certificate_verify(
                                           uint16_t *algorithm )
 {
     uint16_t *sig_alg = ssl->handshake->received_sig_algs;
+    psa_algorithm_t psa_alg = 0;
 
     *algorithm = MBEDTLS_TLS1_3_SIG_NONE;
     for( ; *sig_alg != MBEDTLS_TLS1_3_SIG_NONE ; sig_alg++ )
     {
+        switch( *sig_alg )
```

Comment:
I wonder if it might be worth making this into an auxiliary function. Looks like it's only needed in one place now but might be needed elsewhere in the future? Wdyt @ronald-cron-arm ?

Reply:
Yes I would also prefer for this to be in an auxiliary function.